helix-core/src/comment.rs

```diff
@@ -45,31 +51,32 @@ pub fn toggle_line_comments(doc: &Rope, selection: &Selection, token: Option<&st
     let token = token.unwrap_or("//");
     let comment = Tendril::from(format!("{} ", token));
 
+    let mut lines: Vec<usize> = Vec::new();
+
+    let mut min_next_line = 0;
     for selection in selection {
-        let start = text.char_to_line(selection.from());
-        let end = text.char_to_line(selection.to());
-        let lines = start..end + 1;
-        let (commented, skipped, min) = find_line_comment(&token, text, lines.clone());
+        let start = text.char_to_line(selection.from()).max(min_next_line);
+        let end = text.char_to_line(selection.to()) + 1;
+        lines.extend(start..end);
+        min_next_line = end + 1;
+    }
 
-        changes.reserve((end - start).saturating_sub(skipped.len()));
+    let (commented, to_change, min, margin) = find_line_comment(&token, text, lines);
 
-        for line in lines {
-            if skipped.contains(&line) {
-                continue;
-            }
+    changes.reserve(to_change.len());
```

Comment:
Why not initialize changes here?
```suggestion
    let changes = Vec::with_capacity(to_change.len());
```

Reply:
Ah, you're right. Though the suggested change doesn't remove the original `changes` declaration, and doesn't have a `mut` in the new declaration.

Reply:
Oh, I forgot that.

Reply:
Or we could do collect.

Reply:
Collect what?